model/src/conditions/condition-operators.ts

```diff
@@ -163,7 +163,7 @@ function formatValue(
     return value.toExpression()
   }
 
-  return `'${value.toExpression()}'`
+  return `'${value.toExpression().replace(/'/g, "\\'")}'`
```

Comment:
This PR looks amazing 🙌

Two questions though, shouldn't it be in [`toExpression()`](https://github.com/DEFRA/forms-designer/blob/51948c8d2179c47c0b78fdfb6aafc686172f93d4/model/src/conditions/condition-values.ts#L34-L36) where we add the patch?

```patch
  toExpression() {
-   return this.value
+  return this.value.replace(/'/g, "\\'")
  }
```

And secondly, do we need to escape other JavaScript characters like `[`, `{`?

Reply:
```
function formatValue(
  field: Pick<ComponentDef, 'type'>,
  value: Condition['value']
) {
  if (
    (field.type === ComponentType.DatePartsField &&
      value.type === ConditionType.RelativeDate) ||
    field.type === ComponentType.NumberField ||
    field.type === ComponentType.YesNoField
  ) {
    return value.toExpression()
  }

  return `'${value.toExpression().replace(/'/g, "\\'")}'`
}
```

The issue is less about escaping any character but ensuring we aren't generating an invalid string. In the last line of that code block, we're wrapping the value expression itself in single quotes, therefore any single quotes inside the string need excaping.

Also, we could do this inside `toExpression` but it's only here where it's required (we don't want escaping for `DatePartsField`, `RelativeDate`, `NumberField`, `YesNoField` as they don't get wrapped in single quotes)

Reply:
Yeah, I'd expect `toExpression()` to always be a valid expression

Happy to move things